test/functional/http/get/dapps.js

```diff
@@ -1,460 +1,488 @@
-var modulesLoader = require('../../../common/modulesLoader');
-
-describe('GET /api/dapps', function () {
-
-	function postTransaction (transaction, done) {
-		ws.call('postTransactions', { transaction: transaction }, done, true);
-	}
-
-	before(function (done) {
-		modulesLoader.getDbConnection(function (err, db) {
-			if (err) {
-				return done(err);
-			}
-
-			node.async.every(['dapps', 'outtransfer', 'intransfer'], function (table, cb) {
-				clearDatabaseTable(db, modulesLoader.logger, table, cb);
-			}, done);
-		});
-	});
-
-	before(function (done) {
-		node.async.eachSeries([node.guestbookDapp, node.blockDataDapp], function (dapp, eachSeriesCb) {
-			var transaction = node.lisk.dapp.createDapp(node.gAccount.password, null, dapp);
-			dapp.transactionId = transaction.id;
-			postTransaction(transaction, eachSeriesCb);
-		}, done);
-	});
 
-	before(function (done) {
-		node.onNewBlock(done);
+var sendTransactionPromise = require('../../../common/apiHelpers').sendTransactionPromise;
+var creditAccountPromise = require('../../../common/apiHelpers').creditAccountPromise;
+var getDappPromise = require('../../../common/apiHelpers').getDappPromise;
+var getDappsPromise = require('../../../common/apiHelpers').getDappsPromise;
+var getDappsCategoriesPromise = require('../../../common/apiHelpers').getDappsCategoriesPromise;
+var waitForConfirmations = require('../../../common/apiHelpers').waitForConfirmations;
+
+describe('GET /api/dapps', function () {	
+	
+	var transaction;
+	var transactionsToWaitFor = [];
+
+	var account = node.randomAccount();
+	var dapp1 = node.randomApplication();
+	dapp1.category = 1;
+	var dapp2 = node.randomApplication();
+	dapp2.category = 2;
+	
+	before(function () {
+		var promises = [];
+
+		return creditAccountPromise(account.address, 1000 * node.normalizer)
+			.then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').that.is.not.empty;
+				
+				transactionsToWaitFor.push(res.transactionId);
+				return waitForConfirmations(transactionsToWaitFor);
+			})
+			.then(function (res) {
+				transaction = node.lisk.dapp.createDapp(account.password, null, dapp1);
+				
+				return sendTransactionPromise(transaction);
+			})
+			.then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);
+				transactionsToWaitFor = [];
+				dapp1.id = transaction.id;
+				transactionsToWaitFor.push(res.transactionId);
+			})
+			.then(function (res) {
+				transaction = node.lisk.dapp.createDapp(account.password, null, dapp2);
+
+				return sendTransactionPromise(transaction);
+			})
+			.then(function (res) {
+				node.expect(res).to.have.property('success').to.be.ok;
+				node.expect(res).to.have.property('transactionId').to.equal(transaction.id);	
+				transactionsToWaitFor = [];
```

Comment:
we add `dapp1` transaction to `transactionToWaitFor` but never call `waitForConfirmation` before reassigning `transactionToWaitFor` to empty object.